pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -409,15 +427,21 @@ func routing(ctx context.Context, in *executor.Input, platformProviderName strin
 		in.Logger.Error("Failed to store traffic routing config to metadata store", zap.Error(err))
 	}
 
-	currListenerArns, err := client.GetListenerArns(ctx, primaryTargetGroup)
-	if err != nil {
-		in.LogPersister.Errorf("Failed to get current active listeners: %v", err)
-		return false
-	}
-
-	if err := client.ModifyListeners(ctx, currListenerArns, routingTrafficCfg); err != nil {
-		in.LogPersister.Errorf("Failed to routing traffic to PRIMARY/CANARY variants: %v", err)
-		return false
+	if len(listenerRules) == 0 {
```

Comment:
@moko-poi let's make this easier to follow by removing the if/else condition here and return early (as pipecd code convention)
I think we can make it
```go
if len(listenerRules) != 0 {
   // Do modify listener rules
   return true
}
  // Do modify listener
  return true
```

Reply:
@khanhtc1202 
Thank you for the feedback.

I agree that following the pipecd code convention and using early returns will make the code clearer.
I refactored the if/else condition as you suggested and updated the PR.
[29554b1](https://github.com/pipe-cd/pipecd/pull/4540/commits/29554b1725290064014bcafd28be2a5de344d770)